lib/stats/stats-control.c

```diff
@@ -58,14 +75,24 @@ _reset_counter_if_needed(StatsCluster *sc, gint type, StatsCounterItem *counter,
 }
 
 static void
-_reset_counters(void)
+_reset_counters(GList *counters, gboolean reset_aggregates)
 {
   stats_lock();
-  stats_foreach_counter(_reset_counter_if_needed, NULL, NULL);
+  stats_foreach_counter(_reset_counter_if_needed, counters, NULL);
   stats_unlock();
-  stats_aggregator_lock();
-  stats_aggregator_registry_reset();
-  stats_aggregator_unlock();
+
+  if (reset_aggregates)
+    {
+      stats_aggregator_lock();
+      stats_aggregator_registry_reset();
+      stats_aggregator_unlock();
+    }
+}
+
+void
+stats_control_reset_counters(GList *counters, gboolean reset_aggregates)
+{
+  _reset_counters(counters, reset_aggregates);
```

Comment:
we might want to get rid of this wrapper and rename the function above (_reset_counters) to be stats_control_reset_counters().